apps/arena/native/physics/src/collision_detection/sat.rs

```diff
@@ -306,19 +304,37 @@ fn invalid_axis(
     obstacles: &HashMap<u64, Entity>,
     external_wall: &Entity,
 ) -> bool {
-    let mut obstacle_vector: Vec<Entity> = obstacles.clone().into_values().collect();
-    obstacle_vector.push(external_wall.clone());
+    let obstacle_vector: Vec<Entity> = obstacles.clone().into_values().collect();
+    let external_wall_vector = vec![external_wall.clone()];
+
+    // Check if moved vertex are inside map
+    let current_vertex_moved =
+        Position::add(current_vertex, &Position::mult(&axis, circle.radius * 2.0));
+    let mut current_vertex_point = Entity::new_point(polygon.id, current_vertex_moved);
+    let mut current_vertex_point_collisions =
+        current_vertex_point.collides_with(&external_wall_vector);
+
+    let next_vertex_moved = Position::add(next_vertex, &Position::mult(&axis, circle.radius * 2.0));
+    let mut next_vertex_point = Entity::new_point(polygon.id, next_vertex_moved);
+    let mut next_vertex_point_collisions = next_vertex_point.collides_with(&external_wall_vector);
+
+    if current_vertex_point_collisions.is_empty() && next_vertex_point_collisions.is_empty() {
+        return true;
+    }
```

Comment:
Can you add comments explaining the reason behind all these operations?
It's easy to notice what are you doing code-wise (multiplying a position by player radius, checking a collision, if a list is empty, etc.) but it's pretty hard to know the reason behind each one.

Reply:
Sure thing! i also corrected some typos and updated the whole method comment since it was outdated